cmd/minikube/cmd/service.go

```diff
@@ -67,3 +76,27 @@ func init() {
 	serviceCmd.Flags().BoolVar(&serviceURLMode, "url", false, "Display the kubernetes service URL in the CLI instead of opening it in the default browser")
 	RootCmd.AddCommand(serviceCmd)
 }
+
+// CheckService waits for the specified service to be ready by returning an error until the service is up
+// The check is done by polling the endpoint associated with the service and when the endpoint exists, returning no error->service-online
+func CheckService(namespace string, service string) error {
```

Comment:
Cool, this looks good now. Could you add some unit tests? You might need to refactor it a bit to make it easy to mock out the endpoint status.